src/uwtools/drivers/ungrib.py

```diff
@@ -0,0 +1,175 @@
+                    "prefix": "FILE",
+                },
+            }
+        }
+        path = self._rundir / "namelist.wps"
+        yield self._taskname(str(path))
+        yield asset(path, path.is_file)
+        yield None
+        path.parent.mkdir(parents=True, exist_ok=True)
+        self._create_user_updated_config(
+            config_class=NMLConfig,
+            config_values=d,
+            path=path,
+        )
+
+    @tasks
+    def provisioned_run_directory(self):
+        """
+        Run directory provisioned with all required content.
+        """
+        yield self._taskname("provisioned run directory")
+        yield [
+            self.gribfile_aaa(),
+            self.namelist_wps(),
+            self.runscript(),
+            self.vtable(),
+        ]
+
+    @tasks
+    def run(self):
+        """
+        A run.
+        """
+        yield self._taskname("run")
+        yield (self._run_via_batch_submission() if self._batch else self._run_via_local_execution())
+
+    @task
+    def runscript(self):
+        """
+        The runscript.
+        """
+        path = self._runscript_path
+        yield self._taskname(path.name)
+        yield asset(path, path.is_file)
+        yield None
+        envcmds = self._driver_config.get("execution", {}).get("envcmds", [])
+        execution = [self._runcmd, "test $? -eq 0 && touch %s/done" % self._rundir]
+        scheduler = self._scheduler if self._batch else None
+        path.parent.mkdir(parents=True, exist_ok=True)
+        rs = self._runscript(envcmds=envcmds, execution=execution, scheduler=scheduler)
+        with open(path, "w", encoding="utf-8") as f:
+            print(rs, file=f)
+        os.chmod(path, os.stat(path).st_mode | stat.S_IEXEC)
+
+    @task
+    def vtable(self):
+        """
+        The Vtable.
+        """
+        path = self._rundir / "Vtable.GFS"
+        yield self._taskname(str(path))
+        yield asset(path, path.is_symlink)
+        yield None
+        path.parent.mkdir(parents=True, exist_ok=True)
+        path.symlink_to(Path(self._driver_config["vtable"]))
+
+    # Private helper methods
+
+    @property
+    def _driver_config(self) -> Dict[str, Any]:
+        """
+        Returns the config block specific to this driver.
+        """
+        driver_config: Dict[str, Any] = self._config["ungrib"]
+        return driver_config
+
+    @property
+    def _resources(self) -> Dict[str, Any]:
+        """
+        Returns configuration data for the runscript.
+        """
+        return {
+            "account": self._config["platform"]["account"],
+            "rundir": self._rundir,
+            "scheduler": self._config["platform"]["scheduler"],
+            **self._driver_config.get("execution", {}).get("batchargs", {}),
+        }
+
+    @property
+    def _runscript_path(self) -> Path:
+        """
+        Returns the path to the runscript.
+        """
+        return self._rundir / "runscript"
```

Comment:
I think there's also been an update, available on `main`, that includes the name of the component in the runscript name.